src/org/rocketproplab/marginalstability/flightcomputer/hal/MAX31856.java

```diff
@@ -0,0 +1,121 @@
+package org.rocketproplab.marginalstability.flightcomputer.hal;
+
+import java.io.IOException;
+
+import org.rocketproplab.marginalstability.flightcomputer.Time;
+
+import com.pi4j.io.spi.SpiDevice;
+
+/**
+ * Implements SPI protocol for the MAX31856, contains code to read and quantify
+ * linearized thermocouple temperature data.
+ * 
+ * @author Rudy Thurston
+ *
+ */
+public class MAX31856 implements Thermocouple, PollingSensor {
+
+  private SpiDevice           spi;
+  private double              temp;
+  private long                sampleTime;
+  private Time                clock;
+
+  private static final byte   REG_TC_TEMP = 0x0C;
+  private static final double ZERO_TIME   = 0.0;
+
+  /**
+   * Create instance of MAX31856 with given SpiDevice. Time will be used to
+   * determine the {@link #getLastMeasurementTime()} return value.
+   * 
+   * @param spi       SpiDevice communicating with MAX31856, no validation check
+   * @param time      the time to use when reporting measurement time.
+   */
+  public MAX31856(SpiDevice spi, Time time) {
+	  this.spi = spi;
+	  this.clock = time;
+  }
+
+  @Override
+  public double getTemperature() {
+    return temp;
+  }
+
+  @Override
+  public double getLastMeasurementTime() {
+    if (clock != null) {
+      return sampleTime;
+    } else {
+      return ZERO_TIME;
+    }
+  }
+
+  @Override
+  public void poll() {
+    try {
+      this.readTemp();
+    } catch (IOException e) {
+      // TODO Report error
+      e.printStackTrace();
+    }
+    sampleTime = (long) clock.getSystemTime();
+  }
+
+  /**
+   * Read the linearized thermocouple temperature from the sensor using a
+   * multi-byte read. Read/Write register memory map,
+   * https://datasheets.maximintegrated.com/en/ds/MAX31856.pdf
+   * 
+   * 0x0C (readData[1]) : | Sign | 2^10 | 2^9 | 2^8 | 2^7 | 2^6 | 2^5 | 2^4 |
+   * 0x0D (readData[2]) : | 2^3  | 2^2  | 2^1 | 2^0 | 2^-1| 2^-2| 2^-3| 2^-4|
+   * 0x0E (readData[3]) : | 2^-5 | 2^-6 | 2^-7|  X  |  X  |  X  |  X  |  X  |
+   * 
+   * @throws IOException
+   */
+  private void readTemp() throws IOException {
```

Comment:
I think this has the same pitfall as the other 2's complement conversion method.